content/techniques/hot-reload.md

```diff
@@ -2,17 +2,98 @@
 
 The highest impact on your application's bootstrapping process has a **TypeScript compilation**. But the question is, do we have to recompile a whole project each time when change occurs? Not at all. That's why [webpack](https://github.com/webpack/webpack) HMR (Hot-Module Replacement) significantly decreases an amount of time necessary to instantiate your application.
 
+> warning **Warning** Note that `webpack` won't automatically copy your assets (e.g. `graphql` files) to the `dist` folder. Similary, `webpack` is not compatible with glob static paths (e.g. `entities` property in `TypeOrmModule`).
+
+### With CLI
+
+If you are using [Nest CLI](http://localhost:4200/cli/overview), the configuration process should be pretty straighforward. CLI wraps `webpack` underneath, which means that you only have to apply an additional `HotModuleReplacementPlugin`.
+
 #### Installation
 
-Firstly, let's install required packages:
+First install the required package:
+
+```bash
+$ npm i --save-dev webpack-node-externals
+```
+
+#### Configuration
+
+Once the installation is complete, create a `webpack.config.js` file in the root directory of your application.
+
+```typescript
+const webpack = require('webpack');
+const nodeExternals = require('webpack-node-externals');
+
+export default function(options) {
+  return {
+    ...options,
+    entry: ['webpack/hot/poll?100', './src/main.ts'],
+    watch: true,
+    externals: [
+      nodeExternals({
+        whitelist: ['webpack/hot/poll?100'],
+      }),
+    ],
+    plugins: [...options.plugins, new webpack.HotModuleReplacementPlugin()],
+  };
+}
+```
+
+This function takes the original object containing default webpack configuration and returns a modified one with an applied `HotModuleReplacementPlugin` plugin.
+
+#### Hot-Module Replacement
+
+In order to enable **HMR**, we have to open the application entry file (`main.ts`) and add a few webpack-related instructions.
```

Comment:
use:
In order to enable **HMR**, open the application entry file (`main.ts`) and add several webpack-related instructions, as shown below: